packages/SwingSet/docs/garbage-collection.md

```diff
@@ -48,13 +115,17 @@ function makeStash() {
 }
 ```
 
-We use "RECOGNIZABLE" to mean "recognizable but not reachable", and "REACHABLE" to mean "both recognizable and reachable".
+These predicates must be invoked with a strong reference to an object, so the target must be strongly reachable for the question to even be asked. However, we'll soon introduce the notion of abstract "swingset objects" (as opposed to the normal JavaScript `Object` type), and we want the predicate to answer a question about the abstract identity, even though the question is being asked with the concrete `Object` type. Hence, we must establish the difference between A being able to "reach" B, and A merely being able to "recognize" B.
 
-RECOGNIZABLE happens when the object is used as a key in a `WeakMap` or `WeakSet` entry, but not stored elsewhere. If the vat held a copy of the object to compare against (e.g. with `===`), the object would also be reachable, and we'd be in the REACHABLE state, not RECOGNIZABLE. We do not give userspace code access to the JavaScript `WeakRef` or `FinalizationRegistry` objects, so `WeakMap` and `WeakSet` are the only tools they can use to achieve recognizability without also reachability.
+We use a visual notation in which (strong) reachability is marked with a double line, weak references use a single line, and recognizability is marked with a dotted line:
 
-RECOGNIZABLE is significant because we want a WeakMap in one vat to accept keys which are Presences associated with an exported Remotable from some other vat, and we want to clean up the entries when they go away. We cannot rely upon the JavaScript engine's GC code to perform this cross-vat collection, because the inter-vat links are just data (vrefs and krefs), which are opaque to the JS engine.
+![Reachable vs Recognizable](./images/gc/reachable-vs-recognizable.png "Reachable vs Recognizable")
+
+SwingSet allows vats to use `WeakMap` and `WeakSet` as usual, however the semantics they provide are defined in terms of "swingset objects" instead of strictly using JavaScript `Object` objects.
+
+## JS Engine States: REACHABLE / UNREACHABLE / COLLECTED / FINALIZED / UNKNOWN
```

Comment:
Object states, not engine states.  Well, they're (conceptual) states of the engine's model of an object, but they're not states of the engine itself.

Reply:
Good point.. that title needs to distinguish this section from the later one that talkes about the vat as a whole ("Vat states: REACHABLE vs RECOGNIZABLE"). I'll try "JS Object States: ...".